### ISSUE TITLE
Parser doesn't parse example in doc

### ISSUE THREAD

**Issue by MikeKDowd** [opencog/link-grammar#1282]
The page describing the B link type
https://www.abisource.com/projects/link-grammar/dict/section-B.html

includes this example:

**Noun-Modifying Prepositional-Object Relative Clauses**
A final use of B occurs in relative clauses where the focus of
the clause is the object of a noun-modifying preposition.

                      +----Bpj-----+
               +--MX--+-M--+-Jr+RS-+
               |      |    |   |   |
        The doctors, many of whom are surgeons, were angry

but the parser doesn't parse it:

linkparser> The doctors, many of whom are surgeons, are angry.
No complete linkages found.
Found 6 linkages (6 had no P.P. violations) at null count 2
	Linkage 1, cost vector = (UNUSED=2 DIS= 0.15 LEN=42)

    +----------------------------------Xp---------------------------------+
    +--------------------------->WV-------------------------->+           |
    |             +--------------------Spx--------------------+           |
    |             +---------------MXp---------------+         |           |
    |             |     +-------------Xd------------+         |           |
    +----->Wd-----+     |   +----------Dmc----------+         |           |
    |      +--Dmc-+     |   +OFd+---------Jd--------+--Xc-+   +--Pa--+    |
    |      |      |     |   |   |                   |     |   |      |    |
LEFT-WALL the doctors.n , many of [whom] [are] surgeons.n , are.v angry.a .

(S (NP the doctors.n)
   , many
   (PP (NP (NP of {whom} {are} surgeons.n)
           ,)
       (VP are.v
           (ADJP angry.a)))
   .)

            LEFT-WALL     0.000  hWd+ hWV+ Xp+
                  the     0.000  D+
            doctors.n     0.000  Dmc- Wd- @MXp+ Sp+
                    ,     0.000  Xd+
                 many     0.000  OFd+ Dmc+
                   of     0.000  OFd- Jd+
           surgeons.n     0.100  Jd- Dmc- Xd- MX- Xc+
                    ,     0.000  Xc-
                are.v     0.050  Spx- dWV- Pa+
              angry.a     0.000  Pa-
                    .     0.000  Xp- RW+
           RIGHT-WALL     0.000  RW-

Press RETURN for the next linkage.

All 6 linkages skip "whom" and "are".

**Comment by MikeKDowd** [comment]
This is from unmodified link-grammar-5.10.2

**Comment by linas** [comment]
I just now fixed this in commit 326feb96776ed014978e874c239cc80b76f8ea8b -- it was easy -- the bug was introduced 4 years ago (!!) when tightening up other parts of the grammar. 

The fix will be in version 5.10.4 although it might be a while before that version is available.

**Comment by linas** [comment]
It was broken here:
```
first bad commit
commit dd733a21dca1554676429b2e5f93c3b94cdebccb
Date:   Tue Sep 13 15:03:48 2016 -0500
```
